### PR TITLE
nvidia-nsight-systems 2024.7.1.84-3512561 (new cask)

### PR DESCRIPTION
**Pull request by caic99**
**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

_In the following questions `<cask>` is the token of the cask you're submitting._

After making any changes to a cask, existing or new, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask --online <cask>` is error-free.
- [x] `brew style --fix <cask>` reports no offenses.

Additionally, **if adding a new cask**:

- [x] Named the cask according to the [token reference](https://docs.brew.sh/Cask-Cookbook#token-reference).
- [x] Checked the cask was not [already refused](https://github.com/search?q=repo%3AHomebrew%2Fhomebrew-cask+is%3Aclosed+is%3Aunmerged+&type=pullrequests) (add your cask's name to the end of the search field).
- [x] `brew audit --cask --new <cask>` worked successfully.
- [x] `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --cask <cask>` worked successfully.
- [x] `brew uninstall --cask <cask>` worked successfully.

---

### CODE REVIEW

File: Casks/n/nvidia-nsight-systems.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  name "NVIDIA Nsight Systems"
+  desc "System-wide performance analysis tool"
+  homepage "https://developer.nvidia.com/nsight-systems"
+
+  livecheck do
+    url "https://developer.nvidia.com/tools-downloads.json" # https://developer.nvidia.com/nvidia-cuda-toolkit-developer-tools-mac-hosts
+    strategy :page_match do |page|
+      match = page.match(/NsightSystems-macos-public-(\d+\.\d+\.\d+\.\d+-\d+)\.dmg/i)
+      match[1] if match
+    end
```
Comment (bevanjkay) [inline Casks/n/nvidia-nsight-systems.rb:15]:
```suggestion
  regex(/NsightSystems[._-]macos[._-]public[._-]v?(\d+(?:[.-]d+)+)\.dmg/i)

File: Casks/n/nvidia-nsight-systems.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  end
+
+  depends_on macos: ">= :high_sierra"
+
+  app "NVIDIA Nsight Systems.app"
+
+  # No zap stanza required
```
Comment (bevanjkay) [inline Casks/n/nvidia-nsight-systems.rb:22]:
```suggestion
  # No zap stanza required

```

File: Casks/n/nvidia-nsight-systems.rb (hunk 1 above)
Comment (caic99) [reply Casks/n/nvidia-nsight-systems.rb]: Seems that it doesn't work 😿

File: Casks/n/nvidia-nsight-systems.rb [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  sha256 "9bc4027383649845f8f4ffe5bc9c30aaffbb83167f21c92504ffb8532ea5e018"
+
+  url "https://developer.nvidia.com/downloads/assets/tools/secure/nsight-systems/#{version.major}_#{version.minor}/NsightSystems-macos-public-#{version}.dmg"
+  name "NVIDIA Nsight Systems"
+  desc "System-wide performance analysis tool"
+  homepage "https://developer.nvidia.com/nsight-systems"
+
+  livecheck do
+    url "https://developer.nvidia.com/tools-downloads.json" # https://developer.nvidia.com/nvidia-cuda-toolkit-developer-tools-mac-hosts
+    regex(/NsightSystems[._-]macos[._-]public[._-]v?(\d+(?:[.-]d+)+)\.dmg/i)
```
Comment (daeho-ro) [inline Casks/n/nvidia-nsight-systems.rb:12]:
```suggestion
    regex(/NsightSystems[._-]macos[._-]public[._-]v?(\d+(?:[.-]\d+)+)\.dmg/i)
```

File: Casks/n/nvidia-nsight-systems.rb [hunk 4]
```diff
@@ -0,0 +1,24 @@
+cask "nvidia-nsight-systems" do
+  version "2024.7.1.84-3512561"
+  sha256 "9bc4027383649845f8f4ffe5bc9c30aaffbb83167f21c92504ffb8532ea5e018"
+
+  url "https://developer.nvidia.com/downloads/assets/tools/secure/nsight-systems/#{version.major}_#{version.minor}/NsightSystems-macos-public-#{version}.dmg"
```
Comment (khipp) [inline Casks/n/nvidia-nsight-systems.rb:5]:
```suggestion
  url "https://developer.nvidia.com/downloads/assets/tools/secure/nsight-systems/#{version.major_minor.dots_to_underscores}/NsightSystems-macos-public-#{version}.dmg"
```

File: Casks/n/nvidia-nsight-systems.rb [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  name "NVIDIA Nsight Systems"
+  desc "System-wide performance analysis tool"
+  homepage "https://developer.nvidia.com/nsight-systems"
+
+  livecheck do
+    url "https://developer.nvidia.com/tools-downloads.json" # https://developer.nvidia.com/nvidia-cuda-toolkit-developer-tools-mac-hosts
```
Comment (khipp) [inline Casks/n/nvidia-nsight-systems.rb:11]:
```suggestion
    url "https://developer.nvidia.com/tools-downloads.json"
```

File: Casks/n/nvidia-nsight-systems.rb [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    regex(/NsightSystems[._-]macos[._-]public[._-]v?(\d+(?:[.-]\d+)+)\.dmg/i)
```
Comment (khipp) [inline Casks/n/nvidia-nsight-systems.rb:12]:
```suggestion
    regex(/NsightSystems[._-]macos[._-]public[._-]v?(\d+(?:[.-]\d+)+)\.dmg/i)
    strategy :json do |json, regex|
      json["downloads"]&.map do |download|
        next unless download["development_platform"]&.include?("osx")

        download["files"]&.map do |file|
          match = file["url"]&.match(regex)
          next if match.blank?

          match[1]
        end
      end.flatten
    end
```

We can be more specific by using the `Json` strategy, which is suitable for handling JSON responses.

Comment (caic99) [reply Casks/n/nvidia-nsight-systems.rb:12]: Indeed. I've tried using json strategy but didn't figure it out, for I'm not familiar with ruby. Thanks!

File: Casks/n/nvidia-nsight-systems.rb (hunk 4 above)
Comment (caic99) [reply Casks/n/nvidia-nsight-systems.rb:5]: I wonder if `dots_to_underscores` is a method for cask version? I can't find the docs anywhere.

Comment (khipp) [reply Casks/n/nvidia-nsight-systems.rb]: It's mentioned in the [cask cookbook](https://docs.brew.sh/Cask-Cookbook#version-methods):

Similar to `dots_to_hyphens`, we provide methods for all logical permutations of `{dots,hyphens,underscores}_to_{dots,hyphens,underscores}`. The same applies to `no_dots` in the form of `no_{dots,hyphens,underscores}`, with an extra `no_dividers` that applies all these at once.

**Review by bevanjkay — APPROVED** [review]

**Review by khipp — APPROVED** [review]
Thank you for your first contribution to `homebrew-cask`, @caic99! 🎉

File: Casks/n/nvidia-nsight-systems.rb (hunk 4 above)
Comment (caic99) [reply Casks/n/nvidia-nsight-systems.rb]: 😢 Sorry for not noticing the text below - I was staring on the table

**Pull request merged**